drivers/include/mma8x5x.h

```diff
@@ -0,0 +1,187 @@
+ *              This driver only implements basic functionality (i.e. no support
+ *              for external interrupt pins).
+ *
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for the MMA8x5x accelerometer driver.
+ *
+ * @author      Johann Fischer <j.fischer@phytec.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef MMA8X5X_H
+#define MMA8X5X_H
+
+#include <stdint.h>
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef MMA8X5X_I2C_ADDRESS
+#define MMA8X5X_I2C_ADDRESS           0x1D /**< Accelerometer Default Address */
+#endif
+
+/**
+ * @brief   Devices supported by this driver
+ */
+enum {
+    MMA8X5X_TYPE_MMA8652 = 0x4a,    /**< MMA8652 */
+    MMA8X5X_TYPE_MMA8653 = 0x5a,    /**< MMA8653 */
+    MMA8X5X_TYPE_MMA8451 = 0x1a,    /**< MMA8451 */
+    MMA8X5X_TYPE_MMA8452 = 0x2a,    /**< MMA8452 */
+    MMA8X5X_TYPE_MMA8453 = 0x3a     /**< MMA8453 */
+};
+
+/**
+ * @brief   Available sampling rates
+ */
+enum {
+    MMA8X5X_RATE_800HZ = (0 << 3),  /**< 800 Hz Ouput Data Rate in WAKE mode */
+    MMA8X5X_RATE_400HZ = (1 << 3),  /**< 400 Hz Ouput Data Rate in WAKE mode */
+    MMA8X5X_RATE_200HZ = (2 << 3),  /**< 200 Hz Ouput Data Rate in WAKE mode */
+    MMA8X5X_RATE_100HZ = (3 << 3),  /**< 100 Hz Ouput Data Rate in WAKE mode */
+    MMA8X5X_RATE_50HZ  = (4 << 3),  /**< 50 Hz Ouput Data Rate in WAKE mode */
+    MMA8X5X_RATE_1HZ25 = (5 << 3),  /**< 12.5 Hz Ouput Data Rate in WAKE mode */
+    MMA8X5X_RATE_6HZ25 = (6 << 3),  /**< 6.25 Hz Ouput Data Rate in WAKE mode */
+    MMA8X5X_RATE_1HZ56 = (7 << 3)   /**< 1.56 Hz Ouput Data Rate in WAKE mode */
+};
+
+/**
+ * @brief   Available range options
+ */
+enum {
+    MMA8X5X_RANGE_2G = 0,           /**< +/- 2 g Full Scale Range */
+    MMA8X5X_RANGE_4G = 1,           /**< +/- 4 g Full Scale Range */
+    MMA8X5X_RANGE_8G = 2            /**< +/- 8 g Full Scale Range */
+};
+
+/**
+ * @brief   Named return values
+ */
+enum {
+    MMA8X5X_OK          =  0,       /**< everything was fine */
+    MMA8X5X_DATA_READY  =  1,       /**< new data ready to be read */
+    MMA8X5X_NOI2C       = -1,       /**< I2C communication failed */
+    MMA8X5X_NODEV       = -2,       /**< no MMA8X5X device found on the bus */
+    MMA8X5X_NODATA      = -3        /**< no data available */
+};
+
+/**
+ * @brief   Configuration parameters
+ */
+typedef struct {
+    i2c_t i2c;                  /**< I2C bus the device is connected to */
+    uint8_t addr;               /**< I2C bus address of the device */
+    uint8_t type;               /**< device type */
+    uint8_t rate;               /**< sampling rate to use */
+    uint8_t range;              /**< scale range to use */
+    uint8_t offset[3];          /**< data offset in X, Y, and Z direction */
+} mma8x5x_params_t;
+
+/**
+ * @brief   Device descriptor for MMA8x5x accelerometers
+ */
+typedef struct {
+    mma8x5x_params_t params;    /**< device configuration parameters */
```

Comment:
No, this is done intentional, as copying the fields save one de-referencing step every time we interact with the device. Typical memory vs. runtime optimization compromise...